src/compiler-debugging.md

```diff
@@ -0,0 +1,320 @@
+  |
+  = help: the trait `std::ops::Add<()>` is not implemented for `{integer}`
+
+error: aborting due to previous error
+
+$ # Now, where does the error above come from?
+$ RUST_BACKTRACE=1 \
+    ./build/x86_64-unknown-linux-gnu/stage1/bin/rustc \
+    error.rs \
+    -Z treat-err-as-bug
+error[E0277]: the trait bound `{integer}: std::ops::Add<()>` is not satisfied
+ --> error.rs:2:7
+  |
+2 |     1 + ();
+  |       ^ no implementation for `{integer} + ()`
+  |
+  = help: the trait `std::ops::Add<()>` is not implemented for `{integer}`
+
+error: internal compiler error: unexpected panic
+
+note: the compiler unexpectedly panicked. this is a bug.
+
+note: we would appreciate a bug report: https://github.com/rust-lang/rust/blob/master/CONTRIBUTING.md#bug-reports
+
+note: rustc 1.24.0-dev running on x86_64-unknown-linux-gnu
+
+note: run with `RUST_BACKTRACE=1` for a backtrace
+
+thread 'rustc' panicked at 'encountered error with `-Z treat_err_as_bug', 
+/home/user/rust/src/librustc_errors/lib.rs:411:12
+note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose 
+backtrace.
+stack backtrace:
+  (~~~ IRRELEVANT PART OF BACKTRACE REMOVED BY ME ~~~)
+   7: rustc::traits::error_reporting::<impl rustc::infer::InferCtxt<'a, 'gcx, 
+             'tcx>>::report_selection_error
+             at /home/user/rust/src/librustc/traits/error_reporting.rs:823
+   8: rustc::traits::error_reporting::<impl rustc::infer::InferCtxt<'a, 'gcx, 
+             'tcx>>::report_fulfillment_errors
+             at /home/user/rust/src/librustc/traits/error_reporting.rs:160
+             at /home/user/rust/src/librustc/traits/error_reporting.rs:112
+   9: rustc_typeck::check::FnCtxt::select_obligations_where_possible
+             at /home/user/rust/src/librustc_typeck/check/mod.rs:2192
+  (~~~ IRRELEVANT PART OF BACKTRACE REMOVED BY ME ~~~)
+  36: rustc_driver::run_compiler
+             at /home/user/rust/src/librustc_driver/lib.rs:253
+$ # Cool, now I have a backtrace for the error
+```
+
+## Getting logging output
+[getting-logging-output]: #getting-logging-output
+
+The compiler has a lot of `debug!` calls, which print out logging information
+at many points. These are very useful to at least narrow down the location of
+a bug if not to find it entirely, or just to orient yourself to why a compiler
+is doing a particular thing.
+
+To see the logs, you need to set the `RUST_LOG` environment variable to
+your log filter, e.g. to get the logs for a specific module, you can run the
+compiler as `RUST_LOG=module::path rustc my-file.rs`. The Rust logs are
+powered by [env-logger], and you can look at the docs linked there to see
+the full `RUST_LOG` syntax. All `debug!` output will then appear in
+standard error.
+
+Note that unless you use a very strict filter, the logger will emit a *lot*
+of output - so it's typically a good idea to pipe standard error to a file
+and look at the log output with a text editor.
+
+So to put it together.
+```
```

Comment:
should be ` ```text `

Reply:
again, couple of shell commands and comments, so made it ```bash